e2e/specs/immunizations.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { expect } from '@playwright/test';
+import { type Visit } from '@openmrs/esm-framework';
+import { generateRandomPatient, type Patient, startVisit, deletePatient } from '../commands';
+import { test } from '../core';
+import { ImmunizationsPage } from '../pages';
+
+let patient: Patient;
+let visit: Visit;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+  visit = await startVisit(api, patient.uuid);
+});
+
+test('Add an immunization', async ({ page }) => {
+  const immunizationsPage = new ImmunizationsPage(page);
+  const headerRow = immunizationsPage.immunizationsTable().locator('thead > tr');
+  const immunizationType = immunizationsPage.immunizationsTable().locator('tbody td:nth-child(2)');
+  const vaccinationDate = immunizationsPage.immunizationsTable().locator('tbody td:nth-child(3)');
+
+  await test.step('When I go to the Immunizations page', async () => {
+    await immunizationsPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I click on the `Record immunizations` link', async () => {
+    await page.getByRole('button', { name: /record immunizations/i }).click();
+  });
+
+  await test.step('Then I should be redirected to the Immunization form', async () => {
```

Comment:
It's not really a redirect, though, because we're staying on the same page.

```suggestion
  await test.step('Then I should see the Immunization form launch in the workspace', async () => {
```